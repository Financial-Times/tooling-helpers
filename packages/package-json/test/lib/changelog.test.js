const createChangelog = require('../../src/lib/changelog');

let changelog;

beforeEach(() => {
    changelog = createChangelog();
});

describe('createChangelog', () => {
    test('returns object with methods for manipulating changelog', () => {
        expect(changelog).toEqual(expect.any(Object));
        Object.values(changelog).forEach((method) => {
            expect(method).toEqual(expect.any(Function));
        });
    });
});

describe('createEntry', () => {
    test('return a changelog object', () => {
        const changelogEntry = changelog.createEntry({
            event: 'setField'
        });
        expect(changelogEntry).toEqual(expect.any(Object));
    });
});

describe('changelog object', () => {
    // TODO: Rename
    test('get method returns raw changelog object', () => {
        const changelogEntry = changelog.createEntry({
            event: 'setField'
        });
        expect(changelogEntry.get()).toEqual({
            event: 'setField',
            alreadyExisted: false,
            changeWritten: false,
            previousValue: undefined
        });
    });
    // TODO: Rename
    test('get method returns changelog field', () => {
        const changelogEntry = changelog.createEntry({
            event: 'setField'
        });
        expect(changelogEntry.get('event')).toEqual('setField');
    });
    test('set method changes a field in the changelog entry', () => {
        const changelogEntry = changelog.createEntry({
            event: 'setField',
            meta: {
              field: 'license'
            }
        });
        changelogEntry.set('previousValue', 'MIT');
        expect(changelogEntry.get('previousValue')).toEqual('MIT');
    });
});

describe('getChangelog', () => {
    test('returns an array of raw changelog objects', () => {
        changelog.createEntry({ event: 'setField' });
        const changelogEntries = changelog.get();
        expect(changelogEntries).toEqual(expect.any(Array));
        expect(changelogEntries).toMatchSnapshot();
    });
});