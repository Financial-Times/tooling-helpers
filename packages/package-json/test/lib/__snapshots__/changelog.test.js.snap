// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getChangelog returns an array of raw changelog objects 1`] = `
Array [
  Object {
    "alreadyExisted": false,
    "changeWritten": false,
    "event": "setField",
  },
]
`;
